File: frontend/src/shared/models/flow-search-area/flow-search-area-dto.ts
import type { FlowStepDto } from "@/shared/models/flow-step/flow-step-dto";
import { FlowDto } from "@/shared/models/flow/flow-dto";

export class FlowSearchAreaDto {
  id: number = -1;
  name: string = "";

  locationLeft: number = -1;
  locationTop: number = -1;
  locationToRight: number = -1;
  locationToBottom: number = -1;

  // Flow
  flowId: number = -1;
  flow: FlowDto = new FlowDto();

  flowSteps: FlowStepDto[] = [];

  constructor(data: Partial<FlowSearchAreaDto> = {}) {
    Object.assign(this, {
      ...data,
    });
  }
}

File: backend/Core/Models/Dtos/Database/FlowSeachArea/FlowSearchAreaDto.cs
using System.Collections.ObjectModel;

namespace Core.Models.Dtos
{
    public class FlowSearchAreaDto
    {
        public int Id { get; set; }


        public string Name { get; set; } = string.Empty;

        public int LocationLeft { get; set; }
        public int LocationTop { get; set; }
        public int LocationToRight { get; set; }
        public int LocationToBottom { get; set; }

        // Flow
        public int FlowId { get; set; }
        public FlowDto Flow { get; set; } = null!;

        public IEnumerable<FlowStepDto> FlowSteps { get; set; } = new Collection<FlowStepDto>();
    }
}
